app/models/exchange_rate_currency_rate.rb

```diff
@@ -4,6 +4,8 @@ class ExchangeRateCurrencyRate < Sequel::Model
   RATES_FILE = 'data/exchange_rates/all_rates.csv'.freeze
   SPOT_RATES_FILE = 'data/exchange_rates/all_spot_rates.csv'.freeze
 
+  many_to_one :currency, key: :currency_code, primary_key: :currency_code, class_name: ExchangeRateCurrency
```

Comment:
```suggestion
  many_to_one :exchange_rate_currency, key: :currency_code, primary_key: :currency_code
```

Reply:
Yeah this is probably a good way to go, i guess sometimes i was worried it might end up looking super long in some places. `ExchangeRateCurrencyRate.exchange_rate_currency` but happy either way

Reply:
I'm easy on this really. I just generally expect the name of the method and the name of the thing coming back from the method are the same